src/components/accounts/two_factor/EnableTwoFactor.js

```diff
@@ -0,0 +1,194 @@
+import { Translate } from 'react-localize-redux';
+import TwoFactorOption from './TwoFactorOption';
+import PhoneInput, { isValidPhoneNumber } from 'react-phone-number-input';
+import { validateEmail } from '../../../utils/account';
+import FormButton from '../../common/FormButton';
+import {
+    initTwoFactor,
+    reInitTwoFactor,
+    verifyTwoFactor,
+    deployMultisig,
+    redirectToApp
+} from '../../../actions/account';
+import { useRecoveryMethods } from '../../../hooks/recoveryMethods';
+import EnterVerificationCode from '../EnterVerificationCode'
+import Container from '../../common/styled/Container.css'
+
+const StyledContainer = styled(Container)`
+
+    h4 {
+        margin-top: 40px;
+
+        span {
+            color: #FF585D;
+        }
+    }
+
+    button {
+        margin-top: 50px !important;
+        width: 100% !important;
+    }
+`
+
+export function EnableTwoFactor(props) {
+
+    const dispatch = useDispatch();
+    const account = useSelector(({ account }) => account);
+    const accountId = account.accountId;
+
+    const [initiated, setInitiated] = useState(false);
+    const [option, setOption] = useState('email');
+    const [email, setEmail] = useState('');
+    const [phoneNumber, setPhoneNumber] = useState('');
+    const recoveryMethods = useRecoveryMethods(accountId);
+    const loading = account.actionsPending.some(action => ['INIT_TWO_FACTOR', 'VERIFY_TWO_FACTOR', 'DEPLOY_MULTISIG'].includes(action))
+
+    const method = {
+        kind: `2fa-${option}`,
+        detail: option === 'email' ? email : phoneNumber
+    }
+
+    useEffect(() => { 
+        const email = recoveryMethods.filter(method => method.kind === 'email')[0];
+        const phone = recoveryMethods.filter(method => method.kind === 'phone')[0];
+
+        if (email) {
+            setEmail(email.detail)
+        }
+
+        if (phone) {
+            setPhoneNumber(phone.detail)
+        }
+        
+    }, [recoveryMethods]);
+
+    const handleNext = async () => {
+
+        const { confirmed, error } = await dispatch(initTwoFactor(accountId, method))
+
+        if (!error) {
+            // found recovery method matching 2fa method, so we can deploy the multisig contract
+            if (confirmed) {
+                handleDeployMultisig()
+            } else {
+                // verify code first
+                console.log('init success')
+                setInitiated(true)
+            }
+        }
+    }
+
+    const handleConfirm = async (securityCode) => {
+
+        const { error } = await dispatch(verifyTwoFactor(accountId, securityCode))
+
+        if (!error) {
+            console.log('confirmed security code')
+            // no error so let's deploy contract
+            handleDeployMultisig()
+        }
+    }
+
+    const handleDeployMultisig = async () => {
+
+        try {
+            await dispatch(deployMultisig())
+        } catch(e) {
+            console.log('Not able to deploy multisig')
```

Comment:
not sure what is the purpose of catching error just to log it. All errors already are automatically logged by Redux middleware AFAIK